src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1488,12 +1488,31 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  AppIdURL app_url = policy_manager_->GetNextUpdateUrl(urls);
-  while (!IsUrlAppIdValid(app_url.first, urls)) {
-    app_url = policy_manager_->GetNextUpdateUrl(urls);
+  const uint32_t app_id = GetAppIdForSending();
+  const std::string app_policy_id =
+      application_manager_.application(app_id)->policy_app_id();
```

Comment:
@ZhdanovP potentially dangerous code. Please add check that application was found and valid shared pointer returned

Reply:
@AKalinich-Luxoft  please take a look 72ed6514429a8b4d8b11691bccbbe82365b8cbb9